hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java

```diff
@@ -166,7 +166,7 @@ private String getInventoryId(NamedObject no) {
         } else {
             id = no.getID().getIDString();
         }
-        return feedId + "~" + id;
+        return id;
```

Comment:
So resources always have no.getID() != null, and resource type always have it == null?

Reply:
yes, that is the way things work currently.